src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentPageDirectivePass.cs

```diff
@@ -59,15 +59,16 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
             var pageDirective = (DirectiveIntermediateNode)directives[i].Node;
 
             // The parser also adds errors for invalid syntax, we just need to not crash.
-            var routeToken = pageDirective.Tokens.FirstOrDefault();
+            var routeToken = pageDirective.Tokens.First();
```

Comment:
Yeah, we won't ever actually encounter a page directive without tokens, because the way it parses goes down a different path. I added the `PageDirective_MissingRoute` test to show that behavior.